main/src/main/scala/sbt/internal/Load.scala

```diff
@@ -355,7 +375,9 @@ private[sbt] object Load {
           log.warn(
             s"""The project $project references an unknown configuration "$config" and was guessed to be "$guess"."""
           )
-          log.warn("This configuration should be explicitly added to the project.")
+          log.warn(
+            "This configuration should be explicitly added to the `projectConfigurations` of a plugin or the `configs` of the project."
```

Comment:
In the hindsight neither of the options are actually good advice. Adding a new configuration to the subproject has impact on the build beyond key lookup in the shell, so without understanding what the configuration does I don't think we can tell user to do so.

See for example https://github.com/lightbend/paradox/issues/225 which adds unwanted dependency on POM.

Since #4293, I am learning that people put config-scoped settings for all sorts of reasons, like code formatting on optionally added configuration. In those cases we can pretty much ignore those settings and not much would happen.

Reply:
If someone publishes using Ivy layout `ivy.xml` will list configurations. So that would be another side effect, I think.

So I'd suggest the first step we should take is drop the warning at load time, and release 1.2.1.

Next, we should figure how sbt 0.13 is able to display `web-assets:` when you type `w[tab]` in the shell.